### PR TITLE
Constant patterns with is operator

### PR DESCRIPTION
**Pull request by CharlesTaylor7**
Support for [constant patterns w/ is operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is#constant-pattern).

### CODE REVIEW

**Comment by johnnyreilly** [comment]
So cool to see all the activity here! (I've been itching for prettier c# support 😄)

File: src/csharp/CSharpParser.g4 [hunk 1]
```diff
@@ -392,7 +394,7 @@ simple_embedded_statement
 
 	// selection statements
 	| IF OPEN_PARENS expression CLOSE_PARENS if_body (ELSE if_body)?               #ifStatement
-    | SWITCH OPEN_PARENS expression CLOSE_PARENS OPEN_BRACE switch_section* CLOSE_BRACE           #switchStatement
+  | SWITCH OPEN_PARENS expression CLOSE_PARENS OPEN_BRACE switch_section* CLOSE_BRACE           #switchStatement
```
Comment (warrenseine) [inline src/csharp/CSharpParser.g4]: I'm not sure if indentation is incorrect or just GitHub reviews that make it look incorrect.

File: test/CSharp7Features.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace CSharp7
+{
+  class IsOperator
+  {
+    void IsPatternMatch() {
+      if (3 is int foo) {}
```
Comment (warrenseine) [inline test/CSharp7Features.cs]: Please keep everything in the same (long) `AllInOne.cs` file. No need for a specific C# 7 namespace / file.

**Review by warrenseine — CHANGES_REQUESTED** [review]

**Comment by warrenseine** [comment]
Thanks a lot! As mentioned in another PR, let's merge the PR that removes the generated files first.

**Comment by warrenseine** [comment]
Argh. I tried fixing the tabs conflicts from the UI and GitHub produced a merge commit... I'll rewrite and force-push, sorry for that.

**Comment by warrenseine** [comment]
Hum, GitHub doesn't understand that I updated your commit to resolved conflicts and pushed it to master. I probably skipped a step somewhere. Anyway, let's closed this PR now!

**Pull request closed without merging**